### ISSUE TITLE
Reduce Container Usage

### ISSUE THREAD

**Issue by michaelstaib** [SwissLife-OSS/squadron#106]
At the moment we are using one container per class instance since in most cases we inject a resource as a `IClassFixture`. This can really drag down test performance. In many cases, for the test, it would be enough to create a new database within the container.

So the question is should we have some kind of a runtime state and reuse a container instance through multiple resources. I think the main issue why we did not do that was that we could not determine when the last test has be run and we can get rid of the container. Essentially a test run wide init.

**Comment by michaelstaib** [comment]
Something like this could be a way forward on this:
https://github.com/devlooped/xunit.assemblyfixture

@nscheibe what do you think?

**Comment by glucaci** [comment]
This can be tackled with `ICollectionFixture`.

You can create for example a `ICollectionFixture` for `MongoDB`
```csharp
[CollectionDefinition("MongoContainer")]
public class MongoCollectionFixture : ICollectionFixture<MongoResource> { }
```
and than use `MongoCollectionFixture` in all test classes which needs a `MongoResource`. This will create only one container for all classes which are using it.
```csharp
[Collection("MongoContainer")]
public class RepositoryTests 
{
   public RepositoryTests (MongoResource mongoResource) { }
}
```

**Comment by michaelstaib** [comment]
Collection fixture will not allow for parallel test runs.

**Comment by glucaci** [comment]
Yes, that's right you don't have support for parallel test runs but at least for non-parallel ones that will help you.
[Here](https://github.com/xunit/samples.xunit/blob/main/AssemblyFixtureExample/Samples.cs) is a sample from xUnit how we could do it.

**Comment by michaelstaib** [comment]
@glucaci this looks interesting :) It's much simpler them the repo I posted earlier. Still, the question remains if we want to build this into Squadron. We could register the attributes with the package installation and make it very simple for the consumer. 

Still pondering if we in this case still want to inject the resource just to keep it the same and underneath have a pool that handles the running containers. This would then allow us to either create a container per test/class/assembly.
